diff --git a/node_modules/libp2p-record/src/selection.js b/node_modules/libp2p-record/src/selection.js
index 715c236..ee41915 100644
--- a/node_modules/libp2p-record/src/selection.js
+++ b/node_modules/libp2p-record/src/selection.js
@@ -34,6 +34,8 @@ const bestRecord = (selectors, k, records) => {
     throw errcode(new Error(errMsg), 'ERR_UNRECOGNIZED_KEY_PREFIX')
   }
 
+  if (records.length===1) return 0
+
   return selector(k, records)
 }
 
